### PR TITLE
Feature/users controller

### PR DESCRIPTION
**Pull request by angelicavalenta**
created user controllers

### CODE REVIEW

File: router.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+const router = require('koa-router')();
```
Comment (dbyington) [inline router.js:1]: add 'use strict'

File: index.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const Koa = require('koa');
```
Comment (dbyington) [inline index.js:1]: add 'use strict'

**Review by dbyington — DISMISSED** [review]
This is the a pull that includes the status thus far.
There are perhaps something that should be in separate commit/pulls but due to interdependencies it took a lot of extra changes/files to get to a good point to commit, push, and submit the pull request.
There are a couple "housekeeping" items I commented inline.

**Review by dbyington — COMMENTED** [review]
This is the a pull that includes the status thus far.
There are perhaps something that should be in separate commit/pulls but due to interdependencies it took a lot of extra changes/files to get to a good point to commit, push, and submit the pull request.
There are a couple "housekeeping" items I commented inline.

File: controllers/users.controller.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const User = require('../models/user.model');
+const db = require('../db');
+const _ = require('lodash');
+
+module.exports.handleAuthorizeUser = async (ctx, next) => {
+  const user = await getFilteredUser(ctx.status.accessToken);
```
Comment (arol) [inline controllers/users.controller.js:8]: It sounds like a middleware to me 🤔

File: controllers/users.controller.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  ctx.status = 200;
+  ctx.body = user;
+};
+
+
+module.exports.handleUserGet = async (ctx, next) => {
+  const user = await getFilteredUser(ctx.status.accessToken, filterAccessToken);
+  ctx.status = 200;
+  ctx.body = user;
```
Comment (arol) [inline controllers/users.controller.js:17]: In this case, rely on `getFilteredUser` middleware as usual, and use a serializer to print your user.

`ctx.body = UserSerializer.serialize(user)`

A serializer is a function that takes a model and responds with secure-to-print object representing the instance.

**Review by arol — CHANGES_REQUESTED** [review]

**Comment by dbyington** [comment]
stale

**Pull request closed without merging**